CIP-0031/CIP-0031.md

```diff
@@ -0,0 +1,218 @@
+---
+CIP: 31
+Title: On-Chain Token Metadata Standard
+Authors: Matt Ho <matt.ho@gmail.com>
+Comments-URI: https://github.com/cardano-foundation/CIPs/pull/137
+Status: Draft
+Type: Standards
+Created: 2020-10-12
+License: CC-BY-4.0
+---
+
+# Abstract
+
+This specification defines a new transaction metadata label (i.e. `20`) and entry for [CIP-0031] to associate token metadata 
+during a minting transaction.  Unlike CIP-28, this proposal is intended to define a specific schema to allow basic information
+about fungible tokens to be provided On-Chain.
+
+`20` is proposed as the metadata label in homage to the `ERC20` standard on Ethereum.
+
+# Motivation
+
+The (Cardano Token Registry)[https://github.com/cardano-foundation/cardano-token-registry] is a centralized Off-Chain governance system to manage token 
+metadata.  This proposal aims to provide an On-Chain solution to the same problems that will afford users a much higher degree of trust than the existing
+system.  
+
+Specifically, this proposal aims to solve for the following:
+
+* Provide a standardized way to associate an icon and a ticker with a token
+* Provide a mechanism that allows the viewer to evaluate the trustworthiness and/or authenticity of a token
+
+# Specification
+
+Minters of tokens on the Cardano blockchain main optionally choose to associate the following transaction metadata to facilitate off-chain labeling of tokens.
+When doing so, transaction metadata MUST be included in the same transaction that mints the asset id.
+
+> <sub>NOTE</sub>
+>
+> The following provides an example of the proposed on-chain metadata:
+```json
+{
+  "20": {
+    "<policy-id>": {
+      "<asset-name>": {
+        "ticker": "SUNDAE",
+        "url": "https://dex.sundaeswap.finance/assets/{policy-id}.{asset-name}",
+        "desc": "SUNDAE",
+        "icon": "ipfs://ipfs/<ipfs-hash>",
+        "icon-<size>": "ipfs://ipfs/<ipfs-hash>",
+        "decimals": 6,
+        "version": "1.0"
+      }
+    }
+  }
+}
+```
+
+Field       |          | Description            |
+:---        | :----    | :----                  |
+policy-id   |          | token policy id        |
+asset-name  |          | hex encoded asset name |
+ticker      | OPTIONAL | when present, field and overrides default ticker which is the asset-name |
+url         | OPTIONAL | https only url that refers to metadata stored offchain.  The URL SHOULD use the project domain and MUST return authenticity metadata in either html or json format (see below) |
+desc        | OPTIONAL | additional description that defines the usage of the token |
```

Comment:
Probably it's good to highlight that `desc` and `logo` are different form the CF registry

Reply:
Oops, I accidentally pressed "approve" instead of comment, but I don't oppose this CIP anyway